### DISCUSSION TITLE
Next Major Ruby SDK Version

### DISCUSSION THREAD

**Discussion by Dani-Kirby** [braintree/braintree_ruby#229] [Ideas]
## 👋  Welcome
We're using GitHub Discussions as a place to engage with the members of our community, to share ideas, and to support one another.

## 💡  Discussion Topic: Next Major Version of the Braintree Ruby SDK
The Braintree Ruby SDK team is looking to determine what to include in our next major version release (v5). With this update we are dropping support for Ruby 2.6 and are looking for some insight on:
- Language-specific improvements that you would like to see
- Features you'd like us to consider
- Pain points you'd like to see resolutions to

## 🗣 Let Us Know
Please use this thread to voice your opinions, concerns, or thoughts! We'd love to know how we can make the next version better for you.

**Comment by excid3** [comment]
Would love to see some of the warnings cleaned up. I maintain a Rubygem called [Pay](https://github.com/pay-rails/pay) that supports Braintree and our CI output is always full of warnings from the Braintree gem.

Here's an example from a recent CI run: https://github.com/pay-rails/pay/runs/8135613355?check_suite_focus=true
```ruby
/home/runner/work/pay/pay/vendor/bundle/ruby/2.7.0/gems/braintree-4.8.0/lib/braintree/transaction/google_pay_details.rb:32: warning: instance variable @virtual_card_type not initialized
/home/runner/work/pay/pay/vendor/bundle/ruby/2.7.0/gems/braintree-4.8.0/lib/braintree/transaction/google_pay_details.rb:33: warning: instance variable @virtual_card_last_4 not initialized
/home/runner/work/pay/pay/vendor/bundle/ruby/2.7.0/gems/braintree-4.8.0/lib/braintree/transaction.rb:304: warning: instance variable @local_payment not initialized
/home/runner/work/pay/pay/vendor/bundle/ruby/2.7.0/gems/braintree-4.8.0/lib/braintree/transaction.rb:306: warning: instance variable @paypal not initialized
/home/runner/work/pay/pay/vendor/bundle/ruby/2.7.0/gems/braintree-4.8.0/lib/braintree/transaction.rb:307: warning: instance variable @paypal_here not initialized
/home/runner/work/pay/pay/vendor/bundle/ruby/2.7.0/gems/braintree-4.8.0/lib/braintree/transaction.rb:315: warning: instance variable @venmo_account not initialized
/home/runner/work/pay/pay/vendor/bundle/ruby/2.7.0/gems/braintree-4.8.0/lib/braintree/transaction.rb:296: warning: instance variable @apple_pay not initialized
```
